libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -525,10 +584,27 @@ void AP_Baro::update(void)
         sensors[i].healthy = (now - sensors[i].last_update_ms < 500) && !is_zero(sensors[i].pressure);
     }
 
+    // find current GPS altitude
+    float gps_alt = 0.0f;
+    bool process_gps_alt = false;
+    if (!is_zero(_gps_adj_step) && _gps_calibrated) {
+        float gps_vacc;
+        if (_gps->vertical_accuracy(gps_vacc)) {
+            if (gps_vacc < BARO_MAX_GPS_ACCURACY) {
```

Comment:
This should check that the GPS reading is reasonably current. The failure case on GPS failure could mean a constant reading is present until the driver is unloaded. (Unless I need more coffee and have forgotten something). I'd add an age limit of say 0.4 seconds max on the GPS reading, and otherwise discard it).